subprojects/core/src/main/java/org/gradle/initialization/DefaultProjectDescriptor.java

```diff
@@ -31,6 +31,7 @@
 import java.util.Set;
 
 public class DefaultProjectDescriptor implements ProjectDescriptor, ProjectIdentifier {
+    public static final String INVALID_NAME_IN_INCLUDE_HINT = "If you attempt to include a subproject from a nested directory, please use ':' as project delimiter (and not '/') or set 'projectDir' to modify the physical project path";
```

Comment:
❌ Another common problem will be when users have a root dir with spaces in it. In that case the hint should be to set `rootProject.name` explicitly.